compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/destructure-array-assignment-to-context-var.expect.md

```diff
@@ -30,15 +30,15 @@ import { identity } from "shared-runtime";
 function Component(props) {
   const $ = _c(4);
   let x;
-  if ($[0] !== props.value) {
+  if ($[0] !== props) {
```

Comment:
hmm this seems wrong, `props.value` is unconditionally accessed and should be retained as the dep

Reply:
Good catch! This is a bit of a weird one. The reason we're not inferring `props.value` as unconditionally accessed is because we check mutable ranges before adding to `assumedNonNullObjects`. This is to avoid recording loads within an object's mutable range (e.g. `read(x.y.z); x.y = null;`)
```js
// in function collectNodes
      if (value.kind === "PropertyLoad") {
        const propertyNode = c.declareProperty(...);
        if ( !isMutable(instr, value.object))  // <----- this line
          let curr = propertyNode.parent;
          while (curr != null) {
            assumedNonNullObjects.add(curr);
            curr = curr.parent;
          }
        }
```

The edge case here is that we're inferring a mutable range for `props` due to it being aliased into a context variable and captured / mutated within the lambda.